src/server/dev_checks.ts

```diff
@@ -0,0 +1,255 @@
+import { assertSpyCalls, existsSync, spy } from "./deps.ts";
+import {
+  AppModule,
+  ErrorPageModule,
+  Route,
+  StaticFile,
+  UnknownPageModule,
+} from "$fresh/src/server/types.ts";
+import { knownMethods } from "$fresh/src/server/router.ts";
+import { Plugin } from "$fresh/src/server/mod.ts";
+
+export type CheckFunction = () => CheckResult[];
+
+export enum CheckCategory {
+  ModuleExport = "Module Export",
+  MultipleModules = "Multiple Modules",
+  DuplicateRoutePatterns = "Duplicate Route Patterns",
+  HandlerOrComponent = "Handler or Component",
+  StaticDirectory = "Static Directory",
+  StaticFileConflict = "Static File Conflict",
+  PluginRender = "Plugin Render",
+  PluginRenderAsync = "Plugin RenderAsync",
+  PluginInjectModules = "Plugin Inject Modules",
+}
+
+export type CheckResult = {
+  category: CheckCategory;
+  message: string;
+  link?: string;
+};
+
+/** Asserts that the module has a default export */
+export function assertModuleExportsDefault(
+  module: {
+    url: string;
+    module: AppModule | UnknownPageModule | ErrorPageModule;
+  } | null,
+  moduleName: string,
+): CheckResult[] {
+  if (module && !module.module.default) {
+    return [{
+      category: CheckCategory.ModuleExport,
+      message: `Your ${moduleName} file does not have a default export.`,
+      link: module.url,
+    }];
+  }
+  return [];
+}
+
+/** Asserts that only one module exists with the given name. */
+export function assertSingleModule(
+  routes: Route[],
+  moduleName: string,
+): CheckResult[] {
+  const moduleRoutes = routes.filter((route) =>
+    route.name.includes(moduleName)
+  );
```

Comment:
Does this work for routes like `/foo/_app/bar` or would that classify this as an invalid route?

Reply:
whoops, good catch!
i changed the logic to `.endsWith` and added a test case.